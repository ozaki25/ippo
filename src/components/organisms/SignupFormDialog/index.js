import React from 'react';
import propTypes from 'prop-types';
import { Classes, Dialog } from '@blueprintjs/core';
import SignupForm from 'src/components/organisms/SignupForm';

const styles = {
  dialog: {
    margin: '0 15px',
  },
};

const SignupFormDialog = ({ isOpen, onSubmit, onClose }) => (
  <Dialog
    title="ユーザ登録"
    isOpen={isOpen}
    onClose={onClose}
    style={styles.dialog}
    isCloseButtonShown
  >
    <div className={Classes.DIALOG_BODY}>
      <SignupForm onSubmit={onSubmit} />
    </div>
  </Dialog>
);

SignupFormDialog.displayName = 'SignupFormDialog';

SignupFormDialog.propTypes = {
  isOpen: propTypes.bool.isRequired,
  onSubmit: propTypes.func.isRequired,
  onClose: propTypes.func.isRequired,
};

export default SignupFormDialog;
