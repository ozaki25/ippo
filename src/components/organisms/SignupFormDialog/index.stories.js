import React from 'react';
import { storiesOf } from '@storybook/react';
import { boolean } from '@storybook/addon-knobs';
import { action } from '@storybook/addon-actions';
import SignupFormDialog from '.';

const stories = storiesOf('organisms/SignupFormDialog', module);

const props = {
  isOpen: boolean('isOpen', true),
  onSubmit: action('onSubmit'),
  onClose: action('onClose'),
};

stories.add('通常パターン', () => <SignupFormDialog {...props} />);
