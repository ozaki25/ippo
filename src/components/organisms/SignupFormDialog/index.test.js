import React from 'react';
import { snapshot } from 'test/helpers';
import SignupFormDialog from '.';

const props = {
  isOpen: true,
  onSubmit: jest.fn(),
  onClose: jest.fn(),
};

snapshot('SignupFormDialog/nomal', <SignupFormDialog {...props} />);
