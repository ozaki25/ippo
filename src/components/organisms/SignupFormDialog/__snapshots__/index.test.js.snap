// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot: SigninForm/nomal 1`] = `
<MockRouter
  createHref={[Function]}
  location=""
  params={Object {}}
  path=""
  push={[Function]}
  replace={[Function]}
  url=""
>
  <SigninForm
    buttonText="ボタン"
    onSubmit={[MockFunction]}
  >
    <form>
      <Blueprint3.FormGroup
        label="メールアドレス"
        labelFor="email"
      >
        <div
          className="bp3-form-group"
        >
          <label
            className="bp3-label"
            htmlFor="email"
          >
            メールアドレス
             
            <span
              className="bp3-text-muted"
            />
          </label>
          <div
            className="bp3-form-content"
          >
            <Blueprint3.InputGroup
              id="email"
              large={true}
              name="email"
              onChange={[Function]}
              value=""
            >
              <div
                className="bp3-input-group bp3-large"
              >
                <Blueprint3.Icon />
                <input
                  className="bp3-input"
                  id="email"
                  name="email"
                  onChange={[Function]}
                  style={
                    Object {
                      "paddingRight": 10,
                    }
                  }
                  type="text"
                  value=""
                />
              </div>
            </Blueprint3.InputGroup>
          </div>
        </div>
      </Blueprint3.FormGroup>
      <Blueprint3.FormGroup
        label="パスワード"
        labelFor="pass"
      >
        <div
          className="bp3-form-group"
        >
          <label
            className="bp3-label"
            htmlFor="pass"
          >
            パスワード
             
            <span
              className="bp3-text-muted"
            />
          </label>
          <div
            className="bp3-form-content"
          >
            <Blueprint3.InputGroup
              id="pass"
              large={true}
              name="pass"
              onChange={[Function]}
              type="password"
              value=""
            >
              <div
                className="bp3-input-group bp3-large"
              >
                <Blueprint3.Icon />
                <input
                  className="bp3-input"
                  id="pass"
                  name="pass"
                  onChange={[Function]}
                  style={
                    Object {
                      "paddingRight": 10,
                    }
                  }
                  type="password"
                  value=""
                />
              </div>
            </Blueprint3.InputGroup>
          </div>
        </div>
      </Blueprint3.FormGroup>
      <Blueprint3.Button
        disabled={true}
        fill={true}
        large={true}
        onClick={[Function]}
        text="ボタン"
        type="submit"
      >
        <button
          className="bp3-button bp3-disabled bp3-fill bp3-large"
          disabled={true}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          tabIndex={-1}
          type="submit"
        >
          <Blueprint3.Icon
            key="leftIcon"
          />
          <span
            className="bp3-button-text"
            key="text"
          >
            ボタン
          </span>
          <Blueprint3.Icon
            key="rightIcon"
          />
        </button>
      </Blueprint3.Button>
    </form>
  </SigninForm>
</MockRouter>
`;

exports[`should match snapshot: SignupFormDialog/nomal 1`] = `
<MockRouter
  createHref={[Function]}
  location=""
  params={Object {}}
  path=""
  push={[Function]}
  replace={[Function]}
  url=""
>
  <SignupFormDialog
    isOpen={true}
    onClose={[MockFunction]}
    onSubmit={[MockFunction]}
  >
    <Blueprint3.Dialog
      canOutsideClickClose={true}
      isCloseButtonShown={true}
      isOpen={true}
      onClose={[MockFunction]}
      style={
        Object {
          "margin": "0 15px",
        }
      }
      title="ユーザ登録"
    >
      <Blueprint3.Overlay
        autoFocus={true}
        backdropProps={Object {}}
        canEscapeKeyClose={true}
        canOutsideClickClose={true}
        className="bp3-overlay-scroll-container"
        enforceFocus={true}
        hasBackdrop={true}
        isCloseButtonShown={true}
        isOpen={true}
        lazy={true}
        onClose={[MockFunction]}
        style={
          Object {
            "margin": "0 15px",
          }
        }
        title="ユーザ登録"
        transitionDuration={300}
        transitionName="bp3-overlay"
        usePortal={true}
      >
        <Blueprint3.Portal
          container={
            <body
              class="bp3-overlay-open"
            >
              <div
                class="bp3-portal"
              >
                <div
                  class="bp3-overlay bp3-overlay-open bp3-overlay-scroll-container"
                >
                  <div
                    class="bp3-overlay-backdrop bp3-overlay-appear bp3-overlay-appear-active"
                    tabindex="0"
                  />
                  <div
                    class="bp3-dialog-container bp3-overlay-content bp3-overlay-appear bp3-overlay-appear-active"
                    tabindex="0"
                  >
                    <div
                      class="bp3-dialog"
                      style="margin: 0px 15px;"
                    >
                      <div
                        class="bp3-dialog-header"
                      >
                        <h4
                          class="bp3-heading"
                        >
                          ユーザ登録
                        </h4>
                        <button
                          aria-label="Close"
                          class="bp3-button bp3-minimal bp3-dialog-close-button"
                          type="button"
                        >
                          <span
                            class="bp3-icon bp3-icon-small-cross"
                            icon="small-cross"
                          >
                            <svg
                              data-icon="small-cross"
                              height="20"
                              viewBox="0 0 20 20"
                              width="20"
                            >
                              <desc>
                                small-cross
                              </desc>
                              <path
                                d="M11.41 10l3.29-3.29c.19-.18.3-.43.3-.71a1.003 1.003 0 0 0-1.71-.71L10 8.59l-3.29-3.3a1.003 1.003 0 0 0-1.42 1.42L8.59 10 5.3 13.29c-.19.18-.3.43-.3.71a1.003 1.003 0 0 0 1.71.71l3.29-3.3 3.29 3.29c.18.19.43.3.71.3a1.003 1.003 0 0 0 .71-1.71L11.41 10z"
                                fill-rule="evenodd"
                              />
                            </svg>
                          </span>
                        </button>
                      </div>
                      <div
                        class="bp3-dialog-body"
                      >
                        <form>
                          <div
                            class="bp3-form-group"
                          >
                            <label
                              class="bp3-label"
                              for="name"
                            >
                              名前
                               
                              <span
                                class="bp3-text-muted"
                              />
                            </label>
                            <div
                              class="bp3-form-content"
                            >
                              <div
                                class="bp3-input-group bp3-large"
                              >
                                <input
                                  class="bp3-input"
                                  id="name"
                                  name="name"
                                  style="padding-right: 10px;"
                                  type="text"
                                  value=""
                                />
                              </div>
                            </div>
                          </div>
                          <div
                            class="bp3-form-group"
                          >
                            <label
                              class="bp3-label"
                              for="email"
                            >
                              メールアドレス
                               
                              <span
                                class="bp3-text-muted"
                              />
                            </label>
                            <div
                              class="bp3-form-content"
                            >
                              <div
                                class="bp3-input-group bp3-large"
                              >
                                <input
                                  class="bp3-input"
                                  id="email"
                                  name="email"
                                  style="padding-right: 10px;"
                                  type="text"
                                  value=""
                                />
                              </div>
                            </div>
                          </div>
                          <div
                            class="bp3-form-group"
                          >
                            <label
                              class="bp3-label"
                              for="pass"
                            >
                              パスワード
                               
                              <span
                                class="bp3-text-muted"
                              />
                            </label>
                            <div
                              class="bp3-form-content"
                            >
                              <div
                                class="bp3-input-group bp3-large"
                              >
                                <input
                                  class="bp3-input"
                                  id="pass"
                                  name="pass"
                                  style="padding-right: 10px;"
                                  type="password"
                                  value=""
                                />
                              </div>
                            </div>
                          </div>
                          <button
                            class="bp3-button bp3-disabled bp3-fill bp3-large"
                            disabled=""
                            tabindex="-1"
                            type="submit"
                          >
                            <span
                              class="bp3-button-text"
                            >
                              登録
                            </span>
                          </button>
                        </form>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </body>
          }
        >
          <Portal
            containerInfo={
              <div
                class="bp3-portal"
              >
                <div
                  class="bp3-overlay bp3-overlay-open bp3-overlay-scroll-container"
                >
                  <div
                    class="bp3-overlay-backdrop bp3-overlay-appear bp3-overlay-appear-active"
                    tabindex="0"
                  />
                  <div
                    class="bp3-dialog-container bp3-overlay-content bp3-overlay-appear bp3-overlay-appear-active"
                    tabindex="0"
                  >
                    <div
                      class="bp3-dialog"
                      style="margin: 0px 15px;"
                    >
                      <div
                        class="bp3-dialog-header"
                      >
                        <h4
                          class="bp3-heading"
                        >
                          ユーザ登録
                        </h4>
                        <button
                          aria-label="Close"
                          class="bp3-button bp3-minimal bp3-dialog-close-button"
                          type="button"
                        >
                          <span
                            class="bp3-icon bp3-icon-small-cross"
                            icon="small-cross"
                          >
                            <svg
                              data-icon="small-cross"
                              height="20"
                              viewBox="0 0 20 20"
                              width="20"
                            >
                              <desc>
                                small-cross
                              </desc>
                              <path
                                d="M11.41 10l3.29-3.29c.19-.18.3-.43.3-.71a1.003 1.003 0 0 0-1.71-.71L10 8.59l-3.29-3.3a1.003 1.003 0 0 0-1.42 1.42L8.59 10 5.3 13.29c-.19.18-.3.43-.3.71a1.003 1.003 0 0 0 1.71.71l3.29-3.3 3.29 3.29c.18.19.43.3.71.3a1.003 1.003 0 0 0 .71-1.71L11.41 10z"
                                fill-rule="evenodd"
                              />
                            </svg>
                          </span>
                        </button>
                      </div>
                      <div
                        class="bp3-dialog-body"
                      >
                        <form>
                          <div
                            class="bp3-form-group"
                          >
                            <label
                              class="bp3-label"
                              for="name"
                            >
                              名前
                               
                              <span
                                class="bp3-text-muted"
                              />
                            </label>
                            <div
                              class="bp3-form-content"
                            >
                              <div
                                class="bp3-input-group bp3-large"
                              >
                                <input
                                  class="bp3-input"
                                  id="name"
                                  name="name"
                                  style="padding-right: 10px;"
                                  type="text"
                                  value=""
                                />
                              </div>
                            </div>
                          </div>
                          <div
                            class="bp3-form-group"
                          >
                            <label
                              class="bp3-label"
                              for="email"
                            >
                              メールアドレス
                               
                              <span
                                class="bp3-text-muted"
                              />
                            </label>
                            <div
                              class="bp3-form-content"
                            >
                              <div
                                class="bp3-input-group bp3-large"
                              >
                                <input
                                  class="bp3-input"
                                  id="email"
                                  name="email"
                                  style="padding-right: 10px;"
                                  type="text"
                                  value=""
                                />
                              </div>
                            </div>
                          </div>
                          <div
                            class="bp3-form-group"
                          >
                            <label
                              class="bp3-label"
                              for="pass"
                            >
                              パスワード
                               
                              <span
                                class="bp3-text-muted"
                              />
                            </label>
                            <div
                              class="bp3-form-content"
                            >
                              <div
                                class="bp3-input-group bp3-large"
                              >
                                <input
                                  class="bp3-input"
                                  id="pass"
                                  name="pass"
                                  style="padding-right: 10px;"
                                  type="password"
                                  value=""
                                />
                              </div>
                            </div>
                          </div>
                          <button
                            class="bp3-button bp3-disabled bp3-fill bp3-large"
                            disabled=""
                            tabindex="-1"
                            type="submit"
                          >
                            <span
                              class="bp3-button-text"
                            >
                              登録
                            </span>
                          </button>
                        </form>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            }
          >
            <TransitionGroup
              appear={true}
              childFactory={[Function]}
              className="bp3-overlay bp3-overlay-open bp3-overlay-scroll-container"
              component="div"
              onKeyDown={[Function]}
            >
              <div
                className="bp3-overlay bp3-overlay-open bp3-overlay-scroll-container"
                onKeyDown={[Function]}
              >
                <CSSTransition
                  appear={true}
                  classNames="bp3-overlay"
                  in={true}
                  key=".$__backdrop"
                  onExited={[Function]}
                  timeout={300}
                >
                  <Transition
                    appear={true}
                    enter={true}
                    exit={true}
                    in={true}
                    mountOnEnter={false}
                    onEnter={[Function]}
                    onEntered={[Function]}
                    onEntering={[Function]}
                    onExit={[Function]}
                    onExited={[Function]}
                    onExiting={[Function]}
                    timeout={300}
                    unmountOnExit={false}
                  >
                    <div
                      className="bp3-overlay-backdrop"
                      onMouseDown={[Function]}
                      tabIndex={0}
                    />
                  </Transition>
                </CSSTransition>
                <CSSTransition
                  appear={true}
                  classNames="bp3-overlay"
                  in={true}
                  key=".$.0"
                  onExited={[Function]}
                  timeout={300}
                >
                  <Transition
                    appear={true}
                    enter={true}
                    exit={true}
                    in={true}
                    mountOnEnter={false}
                    onEnter={[Function]}
                    onEntered={[Function]}
                    onEntering={[Function]}
                    onExit={[Function]}
                    onExited={[Function]}
                    onExiting={[Function]}
                    timeout={300}
                    unmountOnExit={false}
                  >
                    <div
                      className="bp3-dialog-container bp3-overlay-content"
                      tabIndex={0}
                    >
                      <div
                        className="bp3-dialog"
                        style={
                          Object {
                            "margin": "0 15px",
                          }
                        }
                      >
                        <div
                          className="bp3-dialog-header"
                        >
                          <Blueprint3.Icon
                            iconSize={20}
                          />
                          <Component>
                            <h4
                              className="bp3-heading"
                            >
                              ユーザ登録
                            </h4>
                          </Component>
                          <Blueprint3.Button
                            aria-label="Close"
                            className="bp3-dialog-close-button"
                            icon={
                              <Blueprint3.Icon
                                icon="small-cross"
                                iconSize={20}
                              />
                            }
                            minimal={true}
                            onClick={[MockFunction]}
                          >
                            <button
                              aria-label="Close"
                              className="bp3-button bp3-minimal bp3-dialog-close-button"
                              onClick={[MockFunction]}
                              onKeyDown={[Function]}
                              onKeyUp={[Function]}
                              type="button"
                            >
                              <Blueprint3.Icon
                                icon={
                                  <Blueprint3.Icon
                                    icon="small-cross"
                                    iconSize={20}
                                  />
                                }
                                key="leftIcon"
                              >
                                <Blueprint3.Icon
                                  icon="small-cross"
                                  iconSize={20}
                                >
                                  <span
                                    className="bp3-icon bp3-icon-small-cross"
                                    icon="small-cross"
                                  >
                                    <svg
                                      data-icon="small-cross"
                                      height={20}
                                      viewBox="0 0 20 20"
                                      width={20}
                                    >
                                      <desc>
                                        small-cross
                                      </desc>
                                      <path
                                        d="M11.41 10l3.29-3.29c.19-.18.3-.43.3-.71a1.003 1.003 0 0 0-1.71-.71L10 8.59l-3.29-3.3a1.003 1.003 0 0 0-1.42 1.42L8.59 10 5.3 13.29c-.19.18-.3.43-.3.71a1.003 1.003 0 0 0 1.71.71l3.29-3.3 3.29 3.29c.18.19.43.3.71.3a1.003 1.003 0 0 0 .71-1.71L11.41 10z"
                                        fillRule="evenodd"
                                        key="0"
                                      />
                                    </svg>
                                  </span>
                                </Blueprint3.Icon>
                              </Blueprint3.Icon>
                              <Blueprint3.Icon
                                key="rightIcon"
                              />
                            </button>
                          </Blueprint3.Button>
                        </div>
                        <div
                          className="bp3-dialog-body"
                        >
                          <SignupForm
                            onSubmit={[MockFunction]}
                          >
                            <form>
                              <Blueprint3.FormGroup
                                label="名前"
                                labelFor="name"
                              >
                                <div
                                  className="bp3-form-group"
                                >
                                  <label
                                    className="bp3-label"
                                    htmlFor="name"
                                  >
                                    名前
                                     
                                    <span
                                      className="bp3-text-muted"
                                    />
                                  </label>
                                  <div
                                    className="bp3-form-content"
                                  >
                                    <Blueprint3.InputGroup
                                      id="name"
                                      large={true}
                                      name="name"
                                      onChange={[Function]}
                                      value=""
                                    >
                                      <div
                                        className="bp3-input-group bp3-large"
                                      >
                                        <Blueprint3.Icon />
                                        <input
                                          className="bp3-input"
                                          id="name"
                                          name="name"
                                          onChange={[Function]}
                                          style={
                                            Object {
                                              "paddingRight": 10,
                                            }
                                          }
                                          type="text"
                                          value=""
                                        />
                                      </div>
                                    </Blueprint3.InputGroup>
                                  </div>
                                </div>
                              </Blueprint3.FormGroup>
                              <Blueprint3.FormGroup
                                label="メールアドレス"
                                labelFor="email"
                              >
                                <div
                                  className="bp3-form-group"
                                >
                                  <label
                                    className="bp3-label"
                                    htmlFor="email"
                                  >
                                    メールアドレス
                                     
                                    <span
                                      className="bp3-text-muted"
                                    />
                                  </label>
                                  <div
                                    className="bp3-form-content"
                                  >
                                    <Blueprint3.InputGroup
                                      id="email"
                                      large={true}
                                      name="email"
                                      onChange={[Function]}
                                      value=""
                                    >
                                      <div
                                        className="bp3-input-group bp3-large"
                                      >
                                        <Blueprint3.Icon />
                                        <input
                                          className="bp3-input"
                                          id="email"
                                          name="email"
                                          onChange={[Function]}
                                          style={
                                            Object {
                                              "paddingRight": 10,
                                            }
                                          }
                                          type="text"
                                          value=""
                                        />
                                      </div>
                                    </Blueprint3.InputGroup>
                                  </div>
                                </div>
                              </Blueprint3.FormGroup>
                              <Blueprint3.FormGroup
                                label="パスワード"
                                labelFor="pass"
                              >
                                <div
                                  className="bp3-form-group"
                                >
                                  <label
                                    className="bp3-label"
                                    htmlFor="pass"
                                  >
                                    パスワード
                                     
                                    <span
                                      className="bp3-text-muted"
                                    />
                                  </label>
                                  <div
                                    className="bp3-form-content"
                                  >
                                    <Blueprint3.InputGroup
                                      id="pass"
                                      large={true}
                                      name="pass"
                                      onChange={[Function]}
                                      type="password"
                                      value=""
                                    >
                                      <div
                                        className="bp3-input-group bp3-large"
                                      >
                                        <Blueprint3.Icon />
                                        <input
                                          className="bp3-input"
                                          id="pass"
                                          name="pass"
                                          onChange={[Function]}
                                          style={
                                            Object {
                                              "paddingRight": 10,
                                            }
                                          }
                                          type="password"
                                          value=""
                                        />
                                      </div>
                                    </Blueprint3.InputGroup>
                                  </div>
                                </div>
                              </Blueprint3.FormGroup>
                              <Blueprint3.Button
                                disabled={true}
                                fill={true}
                                large={true}
                                onClick={[Function]}
                                text="登録"
                                type="submit"
                              >
                                <button
                                  className="bp3-button bp3-disabled bp3-fill bp3-large"
                                  disabled={true}
                                  onKeyDown={[Function]}
                                  onKeyUp={[Function]}
                                  tabIndex={-1}
                                  type="submit"
                                >
                                  <Blueprint3.Icon
                                    key="leftIcon"
                                  />
                                  <span
                                    className="bp3-button-text"
                                    key="text"
                                  >
                                    登録
                                  </span>
                                  <Blueprint3.Icon
                                    key="rightIcon"
                                  />
                                </button>
                              </Blueprint3.Button>
                            </form>
                          </SignupForm>
                        </div>
                      </div>
                    </div>
                  </Transition>
                </CSSTransition>
              </div>
            </TransitionGroup>
          </Portal>
        </Blueprint3.Portal>
      </Blueprint3.Overlay>
    </Blueprint3.Dialog>
  </SignupFormDialog>
</MockRouter>
`;
